package group.networkinventorytask.company.inventory.dto.response;

import java.time.LocalDateTime;
import java.util.List;

public class RouterResponse {
    private Long id;
    private Long SiteId;
    private String hostname;
    private String vendor;
    private String model;
    private String serialNumber;
    private String managementIp;
    private String softwareVersion;
    private String status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private List<ShelfTreeResponse> shelves;

    public List<ShelfTreeResponse> getShelves() {
        return shelves;
    }

    public void setShelves(List<ShelfTreeResponse> shelves) {
        this.shelves = shelves;
    }

    public Long getId() {return id;}

    public void setId(Long id) {this.id = id;}

    public String getHostname() {return hostname;}

    public void setHostname(String hostname) {this.hostname = hostname;}

    public String getVendor() {return vendor;}

    public void setVendor(String vendor) {this.vendor = vendor;}

    public String getModel() {return model;}

    public void setModel(String model) {this.model = model;}

    public String getSerialNumber() {return serialNumber;}

    public void setSerialNumber(String serialNumber) {this.serialNumber = serialNumber;}

    public String getManagementIp() {return managementIp;}

    public void setManagementIp(String managementIp) {this.managementIp = managementIp;}

    public String getSoftwareVersion() {return softwareVersion;}

    public void setSoftwareVersion(String softwareVersion) {this.softwareVersion = softwareVersion;}

    public String getStatus() {return status;}

    public void setStatus(String status) {this.status = status;}

    public LocalDateTime getCreatedAt() {return createdAt;}

    public void setCreatedAt(LocalDateTime createdAt) {this.createdAt = createdAt;}

    public LocalDateTime getUpdatedAt() {return updatedAt;}

    public void setUpdatedAt(LocalDateTime updatedAt) {this.updatedAt = updatedAt;}

    public Long getSiteId() {return SiteId;}

    public void setSiteId(Long siteId) {SiteId = siteId;}
}
